### PR TITLE
Added 14 unit tests to `test_reader.cairo`

### PR DESCRIPTION
**Pull request by Akinbola247**
hi @sparqet , @abdelhamidbakhta , @Sk8erboi84 with reference to this issue #491  i added 14 tests cases in the `test_reader.cairo`, 9 are working as expected but 5 are not. the `fn get_pnl_to_pool_factor` returns 0x0, `fn get_virtual_inventory` also returns 0x0, `get_swap_amount_out` & `get_adl_state` returns an unwrap failed error and `get_swap_price_impact` returns a panicked crates error. i've tagged the functions with @audit tag in the `test_reader.cairo` file.  I'm already looking to the reasons for the errors, but that would mean major changes to the code base. Any suggestions or perhaps am i missing something?.

### CODE REVIEW

**Comment by Akinbola247** [comment]
@sparqet i look forward to hearing from you

**Comment by gaetbout** [comment]
@sparqet could you at this and provide some feedback ? 😄

**Comment by Akinbola247** [comment]
ISSUES
1. the fn get_pnl_to_pool_factor returns 0x0 value 
2. fn get_virtual_inventory also returns 0x0
3. get_swap_amount_out returns usd_delta_exceeds_pool_value error (this is probably a due process issue)
4. get_adl_state returns 0x0
5. get_swap_price_impact returns panicked at crates/cheatnet/src/rpc.rs:88:55:
called Option::unwrap() on a None value

These 2 recently stopped working 
6. get_position_info returns thread 'tokio-runtime-worker' panicked at crates/cheatnet/src/rpc.rs:88:55:
called Option::unwrap() on a None value
7. position_info_list returns thread 'tokio-runtime-worker' panicked at crates/cheatnet/src/rpc.rs:88:55:
called Option::unwrap() on a None value


In Summary 7 added tests passed, 2 returns 0x0 value  1 returns a due process error and  2 failed returning a panicked error  (i’ve commented them in the code base)

**Review by Sk8erboi84 — APPROVED** [review]
